spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceAutoConfiguration.java

```diff
@@ -135,6 +140,39 @@ public Object dataSourceMBean(DataSource dataSource) {
 	}
 
 	/**
+	 * Used for avoiding double registration MBean for HikariDataSource.
+	 */
+	@Configuration
+	@ConditionalOnProperty(prefix = "spring.jmx", name = "enabled", havingValue = "true", matchIfMissing = true)
+	@ConditionalOnClass(HikariDataSource.class)
+	@Conditional(DataSourceAutoConfiguration.DataSourceAvailableCondition.class)
+	protected static class HikariDataSourceJmxConfiguration {
+
+		@Autowired
+		private ApplicationContext ctx;
+
+		@Autowired(required = false)
+		private MBeanExporter mBeanExporter;
```

Comment:
Thanks, I rewrote to use constructor injections and ObjectProvider, but I cant inject `List<HikariDataSource>`, since for excluding bean in MBeanExporter bean name should be used